tests/test_search.py

```diff
@@ -19,3 +25,34 @@ def test_empty_response(client) -> None:
 
     assert validator.validate(xml) is True, validator.error_log
     assert root.getchildren() == []
+
+
+def test_search(client, mocker):
+    es_host = 'http://libero-test-elasticsearch'
+    search.ELASTICSEARCH_HOSTS = [es_host]
+    es_search = mocker.patch('search.search.Elasticsearch.search')
+    es_search.return_value = {
+        'hits': {
+            'hits': [
+                {
+                    '_source': {
+                        'id': 'article1', 'service': 'scholarly-articles'
+                    }
+                }
+            ]
+        }
+    }
+
+    path = get_schema_file_path('item-list.rng')
+    xmlschema_doc = etree.parse(path)
+    validator = isoschematron.Schematron(xmlschema_doc)
+
+    response = client.get('/search?article1')
+    assert response.status_code == 200
+    assert response.content_type == 'application/xml; charset=utf-8'
+
+    xml = etree.parse(BytesIO(response.data))
+    assert validator.validate(xml) is True, validator.error_log
+
+    root = xml.getroot()
+    assert len(root.getchildren()) == 1
```

Comment:
could test 2 items rather than 1 (from 2 to N is trivial but from 1 to 2 is where interesting things can break)

Reply:
added a test to return 4 children.